velox/dwio/common/FileSink.h

```diff
@@ -40,15 +40,16 @@ class FileSink : public Closeable {
     /// Connector properties are required to create a FileSink on FileSystems
     /// such as S3.
     const std::shared_ptr<const Config>& connectorProperties{nullptr};
-    memory::MemoryPool* pool{nullptr};
+    /// FileSink pool must be a leaf pool.
+    const std::shared_ptr<memory::MemoryPool>& leafPool{nullptr};
```

Comment:
Let's just keep it a raw pointer and let caller to take the ownership? The memory pool is thread-safe.